manifests/params.pp

```diff
@@ -10,7 +10,7 @@
   $package_ensure    = 'latest'
   $ui_package_name   = 'consul_ui'
   $ui_package_ensure = 'latest'
-  $version = '0.5.0'
+  $version = hiera('consul::version', '0.5.0')  # ensure defaults resolve from hiera as well
```

Comment:
Something must be weird with your hiera setup then.

We use hiera exactly like you mention:

Eg in hiera
```
consul::version: 0.5.1
```

and in params.pp
```
$version = '0.5.0'
```

and the download url becomes:
```
https://dl.bintray.com/mitchellh/consul/0.5.1_web_ui.zip
```